### ISSUE TITLE
Add parallelization for performance

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-aggregator#33]
In various places (and especially the large area processing feature), the aggregator makes synchronous back-end requests in series. While this was the easiest to set up a proof of concept implementation in the existing openeo_driver framework, it is obviously bad for performance because some request can take quite long (e.g. batch job starting).
A lot can be gained by waiting for back-end responses in parallel.

**Comment by soxofaan** [comment]
duplicate of #28
